plaid/plaid.go

```diff
@@ -1,373 +1,136 @@
-		if err = json.Unmarshal(body, &mfaInter); err != nil {
-			return nil, nil, err
-		}
-		mfaRes := mfaResponse{Type: mfaInter.Type, AccessToken: mfaInter.AccessToken}
-		switch mfaInter.Type {
-		case "device":
-			temp, ok := mfaInter.MFA.(interface{})
-			if !ok {
-				return nil, nil, errors.New("Could not decode device mfa")
-			}
-			deviceStruct, ok := temp.(map[string]interface{})
-			if !ok {
-				return nil, nil, errors.New("Could not decode device mfa")
-			}
-			deviceText, ok := deviceStruct["message"].(string)
-			if !ok {
-				return nil, nil, errors.New("Could not decode device mfa")
-			}
-			mfaRes.Device.Message = deviceText
-
-		case "list":
-			temp, ok := mfaInter.MFA.([]interface{})
-			if !ok {
-				return nil, nil, errors.New("Could not decode list mfa")
-			}
-			for _, v := range temp {
-				listArray, ok := v.(map[string]interface{})
-				if !ok {
-					return nil, nil, errors.New("Could not decode list mfa")
-				}
-				maskText, ok := listArray["mask"].(string)
-				if !ok {
-					return nil, nil, errors.New("Could not decode list mfa")
-				}
-				typeText, ok := listArray["type"].(string)
-				if !ok {
-					return nil, nil, errors.New("Could not decode list mfa")
-				}
-				mfaRes.List = append(mfaRes.List, mfaList{Mask: maskText, Type: typeText})
-			}
-
-		case "questions":
-			questions, ok := mfaInter.MFA.([]interface{})
-			if !ok {
-				return nil, nil, errors.New("Could not decode questions mfa")
-			}
-			for _, v := range questions {
-				q, ok := v.(map[string]interface{})
-				if !ok {
-					return nil, nil, errors.New("Could not decode questions mfa")
-				}
-				questionText, ok := q["question"].(string)
-				if !ok {
-					return nil, nil, errors.New("Could not decode questions mfa question")
-				}
-				mfaRes.Questions = append(mfaRes.Questions, mfaQuestion{Question: questionText})
-			}
+type PkgMetadata struct {
+	Name    string `toml:"name"`
+	Version string `toml:"version"`
+	Branch  string `toml:"branch"`
+}
 
-		case "selections":
-			selections, ok := mfaInter.MFA.([]interface{})
-			if !ok {
-				return nil, nil, errors.New("Could not decode selections mfa")
-			}
-			for _, v := range selections {
-				s, ok := v.(map[string]interface{})
-				if !ok {
-					return nil, nil, errors.New("Could not decode selections mfa")
-				}
-				tempAnswers, ok := s["answers"].([]interface{})
-				if !ok {
-					return nil, nil, errors.New("Could not decode selections answers")
-				}
-				answers := make([]string, len(tempAnswers))
-				for i, a := range tempAnswers {
-					answers[i], ok = a.(string)
-				}
-				if !ok {
-					return nil, nil, errors.New("Could not decode selections answers")
-				}
-				question, ok := s["question"].(string)
-				if !ok {
-					return nil, nil, errors.New("Could not decode selections questions")
-				}
-				mfaRes.Selections = append(mfaRes.Selections, mfaSelection{Answers: answers, Question: question})
-			}
-		}
-		return nil, &mfaRes, nil
+type PkgConfig struct {
```

Comment:
does this need to be exported from the package?

Reply:
⚡️